core/store/migrate/migrations/0229_dynamic_pipeline_runs.sql

```diff
@@ -0,0 +1,48 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE job_pipeline_specs (
+    job_id INT NOT NULL,
+    pipeline_spec_id INT NOT NULL,
+    is_primary BOOLEAN NOT NULL DEFAULT FALSE,
+    CONSTRAINT pk_job_pipeline_spec PRIMARY KEY (job_id, pipeline_spec_id),
+    CONSTRAINT fk_job_pipeline_spec_job FOREIGN KEY (job_id) REFERENCES jobs(id) ON DELETE CASCADE DEFERRABLE,
+    CONSTRAINT fk_job_pipeline_spec_pipeline_spec FOREIGN KEY (pipeline_spec_id) REFERENCES pipeline_specs(id) ON DELETE CASCADE DEFERRABLE
+);
+
+CREATE UNIQUE INDEX idx_unique_job_pipeline_spec_primary_per_job ON job_pipeline_specs(job_id) WHERE is_primary;
+
+-- The moment this runs, we only have one job+pipeline_spec combination per job, complying with the unique index.
+INSERT INTO job_pipeline_specs (job_id, pipeline_spec_id, is_primary)
+SELECT id, pipeline_spec_id, TRUE
+FROM jobs;
+
+ALTER TABLE jobs DROP COLUMN pipeline_spec_id; -- Do we use CASCADE here? Does it have any relationship with other tables?
```

Comment:
@vyzaldysanchez Did you look into this? Can you resolve the comment if so?

Reply:
Working on it.

Reply:
This has a relationship with `pipeline_specs`, which I believe is irrelevant now with the new relationship, so it should be safe to drop.

Reply:
I meant the comment about using CASCADE.

Reply:
I'm still not sure about the cascade here, still... Any thoughts @jmank88 ?